cosmwasm/contracts/outpost/src/execute.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+    msg::{Callback, ExecuteMsg, Wasm, WasmHookExecute},
+    state::CONFIG,
+    ContractError,
+};
+use cosmwasm_std::{Addr, Coin, DepsMut, Response, Timestamp};
+
+// IBC timeout
+pub const PACKET_LIFETIME: u64 = 604_800u64; // One week in seconds
+
+//#[cfg(feature = "callbacks")]
+fn build_callback_memo(callback: Option<Callback>) -> Result<String, ContractError> {
+    match callback {
+        Some(callback) => callback.try_string(),
+        None => Ok(String::new()),
+    }
+}
+
+pub fn execute_swap(
+    deps: DepsMut,
+    own_addr: Addr,
+    now: Timestamp,
+    coin: Coin,
+    user_msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let ExecuteMsg::OsmosisSwap {
+        input_coin,
+        output_denom,
+        receiver,
+        slippage,
+        on_failed_delivery,
+        #[cfg(feature = "callbacks")]
+        callback,
+    } = user_msg;
+    let config = CONFIG.load(deps.storage)?;
+
+    // If the callbacks feature is not active, the variable won't exist. Create it here with the default
+    #[cfg(not(feature = "callbacks"))]
+    let callback = None;
+
+    let next_memo = build_callback_memo(callback)?;
```

Comment:
Should we make this callback setting code a separate function that has one impl w/ feature flag, and another without?

May make this simpler to follow.